### ISSUE TITLE
Error parsing emmet snippets.json

### ISSUE THREAD

**Issue by daveswebdesigns** [Microsoft/vscode#41498]
- VSCode Version: 1.19.1
- OS Version: Zorin OS 12.2 (ubuntu 16.04.1)

Steps to Reproduce:

1. add snippets.json directory path in "emmet.extensionsPath" setting
2. restart vscode

Does this issue occur when all extensions are disabled?: Yes

Hi, when I add a snippets.json file to customise emmet snippets, I just keep getting the error: "error while parsing the file /home/dave/conf/snippets/snippets.json"

The developer console basically has the same error: 
error while parsing the file /home/dave/conf/snippets/snippets.json
e.doShow @ messageService.ts:126

Here's the contents of the file:
    {
        "css": {
            "snippets": {
                "bsbb": "box-sizing: border-box",
                "w": "width: 100%",
                "pr": "position: relative"
            }
        }
    }

Thanks, Dave

**Comment by ramya-rao-a** [comment]
@daveswebdesigns Nothing seems to be wrong with the json that you shared. I tried it both on Mac and Ubuntu. It works just fine.

Is that the complete contents of the file?

**Comment by daveswebdesigns** [comment]
Yeah, that's the total contents. Could it be the file encoding? Maybe I
should reinstall? I'm pretty out of ideas

On 12/01/2018 5:53 PM, "Ramya Rao" <notifications@github.com> wrote:

@daveswebdesigns <https://github.com/daveswebdesigns> Nothing seems to be
wrong with the json that you shared. I tried it both on Mac and Ubuntu. It
works just fine.

Is that the complete contents of the file?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/Microsoft/vscode/issues/41498#issuecomment-357157706>,
or mute the thread

**Comment by ramya-rao-a** [comment]
@daveswebdesigns Am sure there is no need to reinstall :) but we can try that as a last resort.

Do you have any other machines that you can try on?
Try our latest Insiders as well from https://code.visualstudio.com/insiders

**Comment by daveswebdesigns** [comment]
Okay, I've updated to 1.19.2 and installed the insiders v1.20, changed the
json to an empty object,  checked the file encoding (utf-8), checked
permissions and still getting the parse error.

Just now installed 1.19.2 on windows machine and got the same error with
the empty object json! I did not see that coming, I thought that would just
work for sure.

It's really got me stumped. Thanks for all your help though, much
appreciated.



On 13/01/2018 5:54 AM, "Ramya Rao" <notifications@github.com> wrote:

@daveswebdesigns <https://github.com/daveswebdesigns> Am sure there is no
need to reinstall :) but we can try that as a last resort.

Do you have any other machines that you can try on?
Try our latest Insiders as well from https://code.visualstudio.com/insiders

Do you see any errors when the file just has an empty json {}?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/Microsoft/vscode/issues/41498#issuecomment-357313671>,
or mute the thread

**Comment by daveswebdesigns** [comment]
Okay, I've updated to 1.19.2 and installed the insiders v1.20, changed the
json to an empty object,  checked the file encoding (utf-8), checked
permissions and still getting the parse error.

I just tried it on a Windows machine with 1.19.2 and the empty json object
and got the same error! That was a surprise.

It's really got me stumped. Thanks for all your help.



On 13/01/2018 5:54 AM, "Ramya Rao" <notifications@github.com> wrote:

@daveswebdesigns <https://github.com/daveswebdesigns> Am sure there is no
need to reinstall :) but we can try that as a last resort.

Do you have any other machines that you can try on?
Try our latest Insiders as well from https://code.visualstudio.com/insiders

Do you see any errors when the file just has an empty json {}?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/Microsoft/vscode/issues/41498#issuecomment-357313671>,
or mute the thread

**Comment by ramya-rao-a** [comment]
The only next step I can think of is to have debug the module that we use to parse the snippets.json file. If you are up for it, follow the below steps

- Clone https://github.com/Microsoft/vscode-emmet-helper
- `cd vscode-emmet-helper`
- `npm install`
- Replace the file at `/testData/custom-snippets-without-inheritence/snippets.json` with your `snippets.json` file
- Add a breakpoint in the test file at https://github.com/Microsoft/vscode-emmet-helper/blob/master/src/test/emmetHelperTest.ts#L303. 
- Press F5 to start debugging the unit tests

You should now be able to step through the code that fetches the snippets.json file and parses it.
See if you still get the parsing error

**Comment by ramya-rao-a** [comment]
@daveswebdesigns Is this still an issue? Can you try on the latest Insiders? I have made some updates to surface a little more details about the parsing error.

**Comment by daveswebdesigns** [comment]
Hi Ramya, wow, thanks for not giving up on this. I did those last unit
tests you suggested and couldn't track down the issue and had a deadline so
just left it.

So, I just installed the latest insiders. I'm not sure what the issue is
but I tried to launch several times and it was taking like 15 minutes to
launch and then working so slow it was unusable.

So I rebooted and then it launched really fast and my shortcuts worked
straight away!

Really weird to switch from a paid product (sublime text) to a free one,
but the more I use it, the more I like it and now the great support.

Thanks so much for your help!


On 27/01/2018 6:56 AM, "Ramya Rao" <notifications@github.com> wrote:

@daveswebdesigns <https://github.com/daveswebdesigns> Is this still an
issue? Can you try on the latest Insiders? I have made some updates to
surface a little more details about the parsing error.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/Microsoft/vscode/issues/41498#issuecomment-360888262>,
or mute the thread
